### PR TITLE
Link checker

### PR DESCRIPTION
**Pull request by hayleycd**
Adding link checker to OSV-Scanner too.

### CODE REVIEW

**Comment by hayleycd** [comment]
I am going to switch this over to act like a cron tomorrow.

**Comment by hayleycd** [comment]
Now runs on a schedule (8:45am Sydney, 3:45 pm Pacific). 

Also runs on pushes, but only checks modified files. Should not prevent merging.

**Comment by hayleycd** [comment]
Converting to draft until we figure out what it going on in the other repo

File: .github/workflows/link-check-on-push.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Check links on push
+
+on: push
+
```
Comment (hayleycd) [inline .github/workflows/link-check-on-push.yml:4]:
```suggestion
permissions:  # added using https://github.com/step-security/secure-repo
  contents: read
```

File: .github/workflows/link-check.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: Check markdown links on schedule
+
+on: 
+  schedule:
+    - cron: '45 22 * * *'
+
```
Comment (hayleycd) [inline .github/workflows/link-check.yml:6]:
```suggestion
permissions:  # added using https://github.com/step-security/secure-repo
  contents: read
```

**Review by another-rex — APPROVED** [review]
Thanks!

**Pull request merged**